src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/CacheableTransformer.groovy

```diff
@@ -0,0 +1,17 @@
+package com.github.jengelman.gradle.plugins.shadow.transformers
+
+import java.lang.annotation.ElementType
+import java.lang.annotation.Retention
+import java.lang.annotation.RetentionPolicy
+import java.lang.annotation.Target
+
+/**
+ * Marks that a given instance of {@link Transformer} is is compatible with the Gradle build cache.
+ * In other words, it has its appropriate inputs annotated so that Gradle can consider them when
+ * determining the cache key.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@interface CacheableTransformer {
```

Comment:
@ghale 

I mark all built-in transformers cacheable, it would be great if you can take a review for #1210.